src/components/UrqlQuery.re

```diff
@@ -0,0 +1,61 @@
+open UrqlTypes;
+
+[@bs.module "urql"] external query: ReasonReact.reactClass = "Query";
+
+type queryRenderPropsJs('a) = {
+  .
+  "fetching": bool,
+  "data": Js.Nullable.t('a),
+  "error": Js.Nullable.t(UrqlError.t),
+  "executeQuery": Js.Json.t => Js.Promise.t('a),
+};
+
+type queryRenderProps('a) = {
+  fetching: bool,
+  data: option('a),
+  error: option(UrqlError.t),
+  executeQuery: Js.Json.t => Js.Promise.t('a),
+  response: response('a),
+};
+
+let urqlDataToRecord = (result: queryRenderPropsJs('a)) => {
+  let data = result##data |> Js.Nullable.toOption;
+  let error = result##error |> Js.Nullable.toOption;
+
+  let response: response('a) =
+    switch (result##fetching, data, error) {
+    | (true, _, _) => Fetching
+    | (false, Some(data), _) => Data(data)
+    | (false, _, Some(error)) => Error(error)
+    | (false, None, None) => NotFound
+    };
+
+  {
+    fetching: result##fetching,
+    data,
+    error,
+    executeQuery: result##executeQuery,
+    response,
+  };
+};
+
+[@bs.deriving abstract]
+type jsProps = {
+  gqlTemplate: string,
```

Comment:
I don't really like this but am getting namespacing issues between the classname being 'query' and the query prop being named 'query'. Any suggestions @parkerziegler?

Reply:
Ah, I hadn't thought about that. Since the React class `external` is only bound to this module and never exposed externally, let's fallback to your original nomenclature of `queryComponent` and stick with keeping the prop named as `query`. I'll update `mutation` to `mutationComponent` after we merge this to keep it consistent.